### ISSUE TITLE
Redundant `if else` doesn't trigger warning

### ISSUE THREAD

**Issue by Rudxain** [rust-lang/rust#99201]
Given the following code:
```rust
let var = if condition {x} else {x};
```

The current output is:
```
```

Ideally the output should look like:
```
Warning: redundant `if` expression. Consider `let var = x`
```
`rustc` shows warnings when there are unused variables and imports, so why not show a warning for something as trivial as a redundant branch?

I tried searching for a duplicate Issue, but didn't find anything.

Tested in 2021 stable edition

**Comment by 5225225** [comment]
https://github.com/rust-lang/rust-clippy has this, as https://rust-lang.github.io/rust-clippy/master/index.html#if_same_then_else . So if we wanted this lint in rustc itself, it would basically just be taking that lint.

**Comment by Rudxain** [comment]
Thank you for the info, I didn't know Clippy had this. Should I close the issue, or wait for more comments?
